### PR TITLE
[P2P Statements] Add Loading Indicator

### PR DESCRIPTION
**Pull request by nickmurray47**
### Details
Add loading indicator while p2p statements page is loading.

### Fixed Issues

$ [GH_LINK](https://github.com/Expensify/Expensify/issues/198551)

### Tests

1. Follow the testing and setup steps from this [PR](https://github.com/Expensify/App/pull/7663)
2. When you load the statements modal with a lot of transactions, verify a loading indicator is present before the content loads

#### Contributor (PR Author) Checklist
- [x] I made sure to pull `main` before submitting my PR for review
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [x] I ran the tests & they passed on **all platforms**
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms) (tested every platform except for mWeb)
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I added comments when the code was not self explanatory
    - [x] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [x] I followed proper naming convention for platform-specific files (if applicable)
    - [x] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)

#### PR Reviewer Checklist
- [ ] I verified the Author pulled `main` before submitting the PR
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the testing environment is mentioned in the test steps
- [ ] I verified testing steps cover success & fail scenarios (if applicable)
- [ ] I verified tests pass on **all platforms** & I tested again on all platforms
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified comments were added when the code was not self explanatory
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [ ] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)

### QA Steps

1. [Only on web for right now] On prod with an account with wallet transactions (can use my own account here), go to `new.expensify.com/statements/202112` and verify a loading indicator displays before the statements page is loaded.

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screen Shot 2022-02-25 at 11 35 10 AM

Screen Shot 2022-02-25 at 11 40 05 AM

#### Desktop
Screen Shot 2022-02-25 at 11 35 10 AM

Screen Shot 2022-02-25 at 11 40 05 AM

#### iOS
Screen Shot 2022-02-25 at 12 43 30 PM

#### Android
Screen Shot 2022-02-25 at 1 30 52 PM

### CODE REVIEW

**Comment by nickmurray47** [comment]
Ready for review!

File: src/components/WalletStatementModal/index.js [hunk 1]
```diff
@@ -1,24 +1,51 @@
...
+    render() {
+        const authToken = lodashGet(this.props, 'session.authToken', null);
+        return (
+            <>
+                {this.state.isLoading && (
+                    <View style={[styles.flex1, styles.alignItemsCenter, styles.justifyContentCenter]}>
+                        <ActivityIndicator
+                            color={themeColors.spinner}
+                            size="large"
+                            style={[styles.walletStatementModalLoadingIndicator]}
```
Comment (thienlnam) [inline src/components/WalletStatementModal/index.js:30]: We have a component called `FullScreenLoadingIndicator` that would make it so you don't have to include these styles

File: src/components/WalletStatementModal/index.native.js [hunk 2]
```diff
@@ -19,6 +22,14 @@ const WalletStatementModal = (props) => {
                 },
             }}
             incognito // 'incognito' prop required for Android, issue here https://github.com/react-native-webview/react-native-webview/issues/1352
+            startInLoadingState
+            renderLoading={() => (
+                <ActivityIndicator
+                    color={themeColors.spinner}
+                    size="large"
+                    style={[styles.walletStatementModalLoadingIndicatorNative]}
```
Comment (thienlnam) [inline src/components/WalletStatementModal/index.native.js:30]: Same here

**Review by thienlnam — CHANGES_REQUESTED** [review]
Looks good but one comment about the loader which can clean up a lot of the code

File: src/components/WalletStatementModal/index.js (hunk 1 above)
Comment (nickmurray47) [reply src/components/WalletStatementModal/index.js:30]: Ah, was thinking that would cover the whole screen but it does exactly what I need it to do. Will update!

**Comment by nickmurray47** [comment]
Screen Shot 2022-02-25 at 2 23 13 PM

**Review by thienlnam — APPROVED** [review]

**Comment by botify** [comment]
@nickmurray47 looks like this was merged without passing tests. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

**Comment by francoisl** [comment]
@nickmurray47 to clarify, can this only be tested on production? I see a loading indicator if I try on staging, but the app reloads over and over.
If it's production QA only, let's check it off of https://github.com/Expensify/App/issues/7954.

**Pull request merged**